exporters/cmake/OpenSSLConfig.cmake.in

```diff
@@ -0,0 +1,188 @@
+# Generated by OpenSSL
+{-
+   use lib catdir($config{sourcedir}, 'Configurations');
+   use platform;
+   use Data::Dumper;
+
+   our %lib_info = (
+       map { my %x = ();
+             my $lib_static = platform->staticlib($_);
+             my $lib_shared =
+                 (platform->can("sharedlib_simple")
+                  ? platform->sharedlib_simple($_)
+                  : undef)
+                 // platform->sharedlib($_);
+             my $lib_shared_import = platform->sharedlib_import($_);
+             $x{static} = $lib_static if $lib_static;
+             $x{shared} = $lib_shared if $lib_shared;
+             $x{shared_import} = $lib_shared_import if $lib_shared_import;
+             %x ? ( $_ => \%x ) : () }
+       grep { !$unified_info{attributes}->{libraries}->{$_}->{noinst} }
+       @{$unified_info{libraries}}
+   );
+   our $no_static = !$disabled{shared} && !$lib_info{libcrypto}->{static};
+   our $no_shared = $disabled{shared};
+
+   # Unixifying of paths, because that's what CMake understands.
+   # This assumes that the given path is absolute.
+   sub unixify {
+       use File::Spec;
+       use File::Spec::Unix;
+
+       my $path = shift;
+       my $no_file = shift;
+       (my $volume, $directories, $file) = File::Spec->splitpath($path, $no_file // ());
+       $directories = File::Spec::Unix->catdir(File::Spec->splitdir($directories));
+
+       # We know that $volume is ignored by File::Spec::Unix, so we handle it specially
+       # according to the conventions of the platform.
+       $volume =~ s|^(.*?):$|/$1| if $^O eq "vms"; # On VMS, DEV: -> /DEV
+
+       return $volume . File::Spec::Unix->catpath('', $directories, $no_file ? () : $file);
+   }
+   ""
+-}
+# Commands may need to know the format version.
+set(CMAKE_IMPORT_FILE_VERSION 1)
+
+{-
+  # It can be argued that 'cmake_path' would be the better choice here.  However,
+  # cmake_path is relatively new, while get_filename_component has been around
+  # since before CMake 3.0
+-}
+# Set up the import path, so all other import paths are made relative this file
+get_filename_component(_ossl_prefix "${CMAKE_CURRENT_LIST_FILE}" PATH)
+{-
+  # For each component in $OpenSSL::safe::installdata::CMAKECONFIGDIR_REL, have CMake
+  # out the parent directory.
+  my $d = unixify($OpenSSL::safe::installdata::CMAKECONFIGDIR_REL);
+  $OUT = '';
+  $OUT .= 'get_filename_component(_ossl_import_path "${_ossl_import_path}" PATH)' . "\n"
+      foreach (split '/', $d);
+-}
+if(_ossl_prefix STREQUAL "/")
+  set(_ossl_prefix "")
+endif()
+
+{- output_off() unless $disabled{shared}; "" -}
+set(OPENSSL_USE_STATIC_LIBS True)
```

Comment:
IMO *setting* this variable doesn't make much sense.
CMake's `FindOpenSSL.cmake` (i.e. Module mode) uses this as an *input* variable which "may be set to control search behavior". In Config mode, the proper handling would be to return `<Pkg>_FOUND` being true or false depending on a match between the desired type and the available type. This would allow CMake Config mode to continue searching for the package in other locations.
And of course there should be no other side effects before the config is considered matching the users request.